static/app/utils/profiling/flamegraph/flamegraphStateProvider/index.tsx

```diff
@@ -1,31 +1,138 @@
-import {createContext} from 'react';
+import {createContext, useEffect} from 'react';
+import {browserHistory} from 'react-router';
+import {Query} from 'history';
 
+import {DeepPartial} from 'sentry/types/utils';
 import {Rect} from 'sentry/utils/profiling/gl/utils';
 import {makeCombinedReducers} from 'sentry/utils/useCombinedReducer';
+import {useLocation} from 'sentry/utils/useLocation';
 import {
   UndoableReducer,
   UndoableReducerAction,
   useUndoableReducer,
 } from 'sentry/utils/useUndoableReducer';
 
+import {useFlamegraphStateValue} from '../useFlamegraphState';
+
 import {flamegraphPreferencesReducer} from './flamegraphPreferences';
+import {flamegraphProfilesReducer} from './flamegraphProfiles';
 import {flamegraphSearchReducer} from './flamegraphSearch';
 import {flamegraphZoomPositionReducer} from './flamegraphZoomPosition';
 
+// Intersect the types so we can properly guard
+type PossibleQuery =
+  | Query
+  | (Pick<FlamegraphState['preferences'], 'colorCoding' | 'sorting' | 'view' | 'xAxis'> &
+      Pick<FlamegraphState['search'], 'query'>);
+
+function isColorCoding(
+  value: PossibleQuery['colorCoding'] | FlamegraphState['preferences']['colorCoding']
+): value is FlamegraphState['preferences']['colorCoding'] {
+  return (
+    value === 'by symbol name' ||
+    value === 'by library' ||
+    value === 'by recursion' ||
+    value === 'by system / application'
+  );
+}
```

Comment:
All these functions are kind of annoying if we add new options later right? Since typescript won't warn us that these definitions are no longer correct if a new option is added.